### PR TITLE
Reduce docker image build verbosity

### PR DESCRIPTION
**Pull request by g-oikonomou**
This pull reduces the docker image build verbosity. We silence git, wget, npm, pip and unzip.

We try and miserably fail to silence apt-get, because apparently dpkg ignores apt-get's verbosity control. That's on their side not ours, but I still think we should run apt-get with -qq.

Nevertheless, this appears enough to keep our output log below the travis log limit.

#719 unresolved, but this can be considered anyway.

### CODE REVIEW

**Comment by simonduq** [comment]
Maybe one option is to use Docker's `--quiet` option. I remember trying once though and the silent output resulted in the jobs killed by Travis after a timeout. The tool `travis_wait` might help.

File: tools/docker/Dockerfile [hunk 1]
```diff
@@ -1,29 +1,39 @@
...
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF && \
+  echo "deb http://download.mono-project.com/repo/ubuntu xenial main" | sudo tee /etc/apt/sources.list.d/mono-xamarin.list && \
+  apt-get -qq update && \
+  apt-get -qq -y --no-install-recommends install \
+    build-essential doxygen git wget unzip python-serial python-pip \
+    default-jdk ant srecord iputils-tracepath rlwrap \
+    mosquitto mosquitto-clients gdb npm \
+    mono-complete gksu libgtk2.0-0 screen uml-utilities gtk-sharp2 \
+    libcanberra-gtk-module:i386 > /dev/null \
+  && apt-get -qq clean
```
[the remark this answers is not in the document]
Comment (g-oikonomou) [reply tools/docker/Dockerfile]: Thanks I've pushed a commit that does that

**Comment by g-oikonomou** [comment]
Instead of removing renode in its entirety, I have made its build success conditional. I am not convinced that the error is fatal: The build seems to succeed, it's a file copy at the end is what fails and this file is not being built in the first place.

I am ready with this one unless more comments.

**Comment by simonduq** [comment]
Great, look all good.
I'd just give @PiotrZierhoffer a day or two to see if he knows what's going on with Renode here

**Comment by g-oikonomou** [comment]
I have removed the commit that made the renode build optional and this now has passed.

File: tools/docker/Dockerfile [hunk 2]
```diff
@@ -1,29 +1,58 @@
...
+    npm \
+    python-pip \
+    python-serial \
+    rlwrap \
+    screen \
+    srecord \
+    uml-utilities \
+    unzip \
+    wget \
+    > /dev/null \
```
Comment (simonduq) [inline tools/docker/Dockerfile]: so, `-qq` was not enough?

Comment (g-oikonomou) [reply tools/docker/Dockerfile]: Nope, -qq will not suppress those:

```
Preconfiguring packages ...
(Reading database ... 10446 files and directories currently installed.)
Preparing to unpack .../perl-base_5.22.1-9ubuntu0.5_i386.deb ...
Unpacking perl-base (5.22.1-9ubuntu0.5) over (5.22.1-9) ...
Setting up perl-base (5.22.1-9ubuntu0.5) ...
(Reading database ... 10446 files and directories currently installed.)
Preparing to unpack .../libc6_2.23-0ubuntu10_i386.deb ...
Unpacking libc6:i386 (2.23-0ubuntu10) over (2.23-0ubuntu3) ...
Setting up libc6:i386 (2.23-0ubuntu10) ...
Processing triggers for libc-bin (2.23-0ubuntu3) ...
(Reading database ... 10446 files and directories currently installed.)
Preparing to unpack .../libssl1.0.0_1.0.2g-1ubuntu4.13_i386.deb ...
Unpacking libssl1.0.0:i386 (1.0.2g-1ubuntu4.13) over (1.0.2g-1ubuntu4.2) ...
Preparing to unpack .../python3.5_3.5.2-2ubuntu0~16.04.4_i386.deb ...
Unpacking python3.5 (3.5.2-2ubuntu0~16.04.4) over (3.5.2-2~16.01) ...
Preparing to unpack .../libpython3.5-stdlib_3.5.2-2ubuntu0~16.04.4_i386.deb ...
Unpacking libpython3.5-stdlib:i386 (3.5.2-2ubuntu0~16.04.4) over (3.5.2-2~16.01) ...
Preparing to unpack .../python3.5-minimal_3.5.2-2ubuntu0~16.04.4_i386.deb ...
Unpacking python3.5-minimal (3.5.2-2ubuntu0~16.04.4) over (3.5.2-2~16.01) ...
Preparing to unpack .../libpython3.5-minimal_3.5.2-2ubuntu0~16.04.4_i386.deb ...
Unpacking libpython3.5-minimal:i386 (3.5.2-2ubuntu0~16.04.4) over (3.5.2-2~16.01) ...
Preparing to unpack .../zlib1g_1%3a1.2.8.dfsg-2ubuntu4.1_i386.deb ...
Unpacking zlib1g:i386 (1:1.2.8.dfsg-2ubuntu4.1) over (1:1.2.8.dfsg-2ubuntu4) ...
Processing triggers for libc-bin (2.23-0ubuntu3) ...
Processing triggers for mime-support (3.59ubuntu1) ...
Setting up zlib1g:i386 (1:1.2.8.dfsg-2ubuntu4.1) ...
Processing triggers for libc-bin (2.23-0ubuntu3) ...
Selecting previously unselected package libgdbm3:i386.
```

and those generate way too much noise in the build

Comment (simonduq) [reply tools/docker/Dockerfile]: ok

**Pull request merged**